### PR TITLE
Update to nb-javac identical to javac from jdk-17+35

### PR DESCRIPTION
**Pull request by jtulach**
Another step in implementation of [automatic nb-javac](https://cwiki.apache.org/confluence/display/NETBEANS/Overview%3A+nb-javac) plan. This PR updates NetBeans to use [nb-javac@jdk-17+35](https://github.com/JaroslavTulach/nb-javac/tree/jdk-17+35) generated automatically from JDK sources and released at [Maven central](https://repo1.maven.org/maven2/com/dukescript/nbjavac/nb-javac/jdk-17+35/).

This PR removes `java.source.nbjavac` module from the distribution and from the sources. Alas the module is still available on update center and gets downloaded unless one disables the _NetBeans Distribution_ update center:

obrazek

Make sure that when a new project is created, it is downloading just three modules, not more:

obrazek

Then Java editing seems to work. If you get into the situation where `java.source.nbjavac` gets installed, you can remove it manually with:
```bash
netbeans$ find nbbuild/netbeans | grep java.source.nbjavac  | xargs rm
netbeans$ touch nbbuild/netbeans/java/.lastModified
```
This problem shall be fixed automatically as soon as this PR is merged. Then `java.source.nbjavac` disappears from the _NetBeans Distribution_ update center.

CCing @mbien

### CODE REVIEW

**Review by matthiasblaesing — DISMISSED** [review]
I build this PR and ran it on OpenJDK 11. There are no code completions and  I see this error in the log:

```
     [exec] java.lang.NoClassDefFoundError: com/sun/tools/javac/api/DuplicateClassChecker
     [exec] 	at org.netbeans.modules.java.source.nbjavac.parsing.TreeLoader$DuplicateClassRegistryImpl.enhance(TreeLoader.java:589)
     [exec] 	at org.netbeans.modules.java.source.parsing.JavacParser.lambda$createJavacTask$13(JavacParser.java:1008)
     [exec] 	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655)
     [exec] 	at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:658)
     [exec] 	at org.netbeans.modules.java.source.parsing.JavacParser.createJavacTask(JavacParser.java:1008)
     [exec] 	at org.netbeans.modules.java.source.parsing.JavacParser.createJavacTask(JavacParser.java:839)
     [exec] 	at org.netbeans.modules.java.source.parsing.CompilationInfoImpl.getJavacTask(CompilationInfoImpl.java:428)
     [exec] 	at org.netbeans.modules.java.source.parsing.CompilationInfoImpl.getJavacTask(CompilationInfoImpl.java:410)
     [exec] 	at org.netbeans.api.java.source.TreeUtilities.<init>(TreeUtilities.java:144)
``` 

The full log
[messages.zip](https://github.com/apache/netbeans/files/7420604/messages.zip)
:

File: java/java.source.base/src/org/netbeans/modules/java/source/indexing/VanillaCompileWorker.java [hunk 1]
```diff
@@ -405,6 +405,7 @@ public void run() throws IOException {
                 JavaIndex.LOG.log(Level.FINEST, "VanillaCompileWorker was canceled in root: " + FileUtil.getFileDisplayName(context.getRoot()), ca);  //NOI18N
             }
         } catch (Throwable t) {
+            t.printStackTrace();
```
[the remark this answers is not in the document]
Comment (JaroslavTulach) [reply java/java.source.base/src/org/netbeans/modules/java/source/indexing/VanillaCompileWorker.java]: Done in 55b99a1

File: nbbuild/cluster.properties [hunk 2]
```diff
@@ -750,6 +749,8 @@ validation.nb.cluster.java=\
         java.hints,\
         java.kit
 
+#        java.source.nbjavac,\
```
[the remark this answers is not in the document]
Comment (JaroslavTulach) [reply nbbuild/cluster.properties]: Removed in 26540ff

**Comment by JaroslavTulach** [comment]
Green! All the gates are green. Let's integrate and move NetBeans towards using the latest JDK `javac`!

**Comment by JaroslavTulach** [comment]
> This problem shall be fixed automatically as soon as this PR is merged. 
> Then `java.source.nbjavac` disappears from the _NetBeans Distribution_ update center.

I've just built new, fresh version from 92c782a8e3bb22, made sure there is no
```bash
netbeans$ find nbbuild/ | grep java.source.nbjavac
```
and then `ant tryme`. Creation of new Maven project succeeded, I was able to create a `Main` class and get code completion. No `java.source.nbjavac` gets downloaded. Looks like the above assumption was correct and we are good to go.

**Pull request merged**